<!DOCTYPE html>
<html lang="en">
<head>
    <meta charset="UTF-8">
    <title>Title</title>
</head>
<body>
<div id="box"></div>
</body>
</html>
<script>
    function loadImg(url) {
        return new Promise((resolve,reject)=>{
            let img=new Image();
            img.src=url;
            img.onload=function () {
                resolve(img)
            };
            img.onerror=function (e) {
                reject(e)
            }
        })
    }
    loadImg("http://img5.imgtn.bdimg.com/it/u=2799293021,1713737542").then((img)=>{
        box.appendChild(img)
        let a1=1
        let a1=1
    }).catch(e=>{
        //捕获错误的，如果new Promise中有错误会被捕获，如果then中的回调有错误也会被捕获 一般最后会加一个catch
        console.log(e);
    })
</script>
